File: php/features.php
<?php
declare(strict_types=1);

// ManualPrecipitationStations SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ManualPrecipitationStationsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ManualPrecipitationStationsBaseFeature();
            case "test":
                return new ManualPrecipitationStationsTestFeature();
            default:
                return new ManualPrecipitationStationsBaseFeature();
        }
    }
}

File: py/features.py
# ManualPrecipitationStations SDK feature factory

from feature.base_feature import ManualPrecipitationStationsBaseFeature
from feature.test_feature import ManualPrecipitationStationsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ManualPrecipitationStationsBaseFeature(),
        "test": lambda: ManualPrecipitationStationsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
